### DISCUSSION TITLE
want to make a fetch call during build

### DISCUSSION THREAD

**Discussion by javag97** [nuxt/nuxt#16803] [Questions]
I have a Vue component that wants to use `$fetch` on a file I created in `/server/api` that gathers the URLs of images in my image directory and lays them out. 

```js
const request = await $fetch("/api/getImages");
```

```typescript
import * as fs from 'fs';
let images = [];

fs.readdir('./public/images', function (err, files) {
    if (err) {
        return console.log('Unable to scan directory: ' + err);
    } 
    files.forEach(function (file) {
        images.push(file);
    });
});

export default defineEventHandler(event => {
  return {
    images: images
  }
});

```

Is there a better way to accomplish this? Or am I doing something wrong? This app works while running with a live Node.js server, but would want this to be statically built so I can host without needing a server.

**Comment by sevillaarvin** [comment]
Hi, see if this helps: https://github.com/nuxt/framework/issues/1693
